src/protocols/aeternity/libs/AeternityAdapter.ts

```diff
@@ -169,8 +174,121 @@ export class AeternityAdapter extends BaseProtocolAdapter {
     // TODO
   }
 
-  async fetchTransactions(): Promise<any> {
-    // TODO
+  async fetchTransactionsFromMiddleware(
+    address: string,
+    nextPageUrl: string | null,
+    limit: number,
+  ) {
+    const {
+      fetchFromMiddlewareCamelCased,
+    } = useMiddleware();
+
+    const url = ([null, ''].includes(nextPageUrl))
+      ? `/v2/accounts/${address}/activities?limit=${limit}`
+      : nextPageUrl!;
+
+    try {
+      const { data, next } = await fetchFromMiddlewareCamelCased(url);
+
+      return {
+        regularTransactions: data || [],
+        nextPageParams: next,
+      };
+    } catch (error) {
+      return {};
+    }
+  }
+
+  async fetchPendingTransactions(
+    address: string,
+    sdk: AeSdkSuperhero,
```

Comment:
I think this method should follow the same pattern as other methods, so the `sdk` should be retrieved in the body of the function instead of passed as the argument.